back/src/controllers/categories/GetCategories.go

```diff
@@ -30,10 +30,11 @@ func GetCategories(ctx *fiber.Ctx) error {
 func queryCategories(page int, ctx *fiber.Ctx) ([]model.Category, error) {
 	pagLimit := constants.PaginationLimit
 	offset := (page - 1) * pagLimit
+	db := config.Database
 
 	categories := make([]model.Category, pagLimit)
-	errQuery := config.Database.Limit(pagLimit).Offset(offset).Order("order_of_display asc").Find(&categories)
-	if errQuery.Error != nil {
+	err := db.Limit(pagLimit).Preload("Products").Offset(offset).Order("order_of_display asc").Find(&categories)
```

Comment:
Why are you preloading "Products", aren't you returning categories ?

Reply:
Preloading products allows to have the products with all the details in the response. It's like with graph fetch ok Objection

Reply:
Sounds good, thanks